gateway/src/main/java/org/georchestra/gateway/security/oauth2/OAuth2Configuration.java

```diff
@@ -83,8 +72,13 @@ public static final class OAuth2AuthenticationCustomizer implements ServerHttpSe
         }
     }
 
-    @Bean
-    private ServerLogoutSuccessHandler oidcLogoutSuccessHandler(
+    /**
+     * TODO: REVISIT, we don't know why this bean has been added, but it breaks
+     * OAuth2SecurityAutoConfigurationTest as there's no
+     * InMemoryReactiveClientRegistrationRepository
```

Comment:
For FranceConnect, a french OpenID provider, we need to send a request to their logout endpoint when user logs out of gateway, so that FranceConnect can also log out the user in their systems. If we do not do that, FranceConnect will refuse any further authentication with a cryptic error until endpoint is called or cookies are cleared.

Spring Security can handle logout endpoints, but only when the OpenID provider provides a discovery endpoint. Then Spring reads the logout endpoint in the response from discovery endpoint, stores it in a map named `configurationMetadata` of `ProviderDetails` class, and uses it further when the user logs out.

But FranceConnect does not provide a discovery endpoint, we need to configure each parameter by hand in YAML file, and there is no way to specify logout endpoint. So we had to create a comprehensive YAML syntax for this, and enumerate all providers to see if it has this parameter set, then add it to `configurationMetadata`using reflection as it is not public and it has no setter. That's what this handler does.

Reply:
Thanks @emmdurin I'll keep that in mind. May ask you for further details once I get to it.

Reply:
You can ask me things about that of course.

Reply:
Just to say that we also need the `setPostLogoutRedirectUri` to get redirected back to gateway after logging out from the provider. So even if the upper part of the method could be implemented elsewhere, we still need this bean.

Reply:
Incorporated as part of #71: https://github.com/georchestra/georchestra-gateway/pull/72/files#diff-13ff067a979409f92be3ebeef5622132d311dbe0cebaf1dbe80ac7a7da97ea26R102